src/app/adminLayout/students/add-student/add-student.component.ts

```diff
@@ -0,0 +1,47 @@
+import { Component } from '@angular/core';
+import {StudentService} from "../../../services/student.service";
+import {FormBuilder} from "@angular/forms";
+import {Student} from "../../../model/student";
+
+@Component({
+  selector: 'app-add-student',
+  templateUrl: './add-student.component.html',
+  styleUrls: ['./add-student.component.css']
+})
+export class AddStudentComponent {
+  checked: boolean;
+  displayModal: boolean;
+  stateOptions: any[];
+  addForm = this.formBuilder.group({
+    firstName: '',
+    lastName: '',
+    email: '',
+    active: true
+  });
+
+  constructor(
+    private studentService: StudentService,
+    private formBuilder: FormBuilder,
+  ) {
+    this.checked=true;
+    this.displayModal=false;
+    this.stateOptions = [
+      {label: 'tak', value: true},
+      {label: 'nie', value: false}
+    ];
+  }
+  onSubmit():void{
+    console.log(this.addForm.value);
+    this.studentService.addStudent(this.addForm.value as Student).subscribe();
```

Comment:
Wytłumaczysz to subscribe?

Reply:
W skrócie: żeby ta metoda addStudent z serwisu została wykonana. 
Ogólnie wszystkie te metody z serwisu zwracają jakiegoś observabla (taki delete też zwraca response który jest observablem), a jak jest zwracany to trzeba go subscribować, bo inaczej ten request nie pójdzie, zgodnie z https://angular.io/guide/http

For all HttpClient methods, the method doesn't begin its HTTP request until you call subscribe() on the observable the method returns.

This is true for all HttpClient methods.

Reply:
A, czyli celowo nic z nim nie robisz, luz.